### ISSUE TITLE
Packer failing just before image creation for windows 2019 on AWS

### ISSUE THREAD

**Issue by exolain** [hashicorp/packer-plugin-amazon#306]
#### Overview of the Issue

Windows 2019 packer json code is no longer working as expected. This same code was working a few days ago but not anymore. Not sure if the AWS marketplace might have added something that is conflicting with the packer script at the end.

I see that it errors out when it runs a packer script:

> 2022/12/23 21:14:26 packer-builder-amazon-ebs plugin: [INFO] command 'powershell -executionpolicy bypass "& { if (Test-Path variable:global:ProgressPreference){set-variable -name variable:global:ProgressPreference -value 'SilentlyContinue'};. c:/Windows/Temp/packer-ps-env-vars-63a6128f-e49a-27c1-0287-fc7740bd31d0.ps1; &'c:/Windows/Temp/script-63a6128f-9916-c398-29e4-929d1f06a84f.ps1'; exit $LastExitCode }"' exited with code: 1
> 2022/12/23 21:14:26 packer-builder-amazon-ebs plugin: [INFO] RPC endpoint: Communicator ended with: 1
> 2022/12/23 21:14:26 [INFO] 374 bytes written for 'stdout'
> 2022/12/23 21:14:26 [INFO] 0 bytes written for 'stderr'
> 2022/12/23 21:14:26 [INFO] RPC client: Communicator ended with: 1

#### Reproduction Steps

Steps to reproduce this issue:
Trying to build a Windows 2019 image using the amazon-ebs builder.
It is failing after the last provisioner stage:
```
        {
            "type": "powershell",
            "inline": [
                "C:\\ProgramData\\Amazon\\EC2-Windows\\Launch\\Scripts\\InitializeInstance.ps1 -SchedulePerBoot",
                "C:\\ProgramData\\Amazon\\EC2-Windows\\Launch\\Scripts\\SysprepInstance.ps1 -NoShutdown"
            ]
        }
```
And I can see that both commands are run successfully:

> 2022/12/23 21:13:04 packer-provisioner-powershell plugin: Found command: C:\ProgramData\Amazon\EC2-Windows\Launch\Scripts\InitializeInstance.ps1 -SchedulePerBoot
> 2022/12/23 21:13:04 packer-provisioner-powershell plugin: Found command: C:\ProgramData\Amazon\EC2-Windows\Launch\Scripts\SysprepInstance.ps1 -NoShutdown
> ==> amazon-ebs: Provisioning with powershell script: /tmp/powershell-provisioner195334040
> 2022/12/23 21:13:04 packer-provisioner-powershell plugin: Opening /tmp/powershell-provisioner195334040 for reading
> 2022/12/23 21:13:04 packer-provisioner-powershell plugin: Uploading env vars to c:/Windows/Temp/packer-ps-env-vars-63a6128f-e49a-27c1-0287-fc7740bd31d0.ps1

But it seems to fail when running the packer that I mention on the description

### Plugin and Packer version

From `packer version: 1.8.5`

### Simplified Packer Buildfile

Gist is blocked by our company, but I have reduced the code with only the relevant parts.

```
  "builders": [{ 
        "type": "amazon-ebs", 
        "region": "{{user `aws_region`}}", 
        "instance_type": "t2.medium", 
        "encrypt_boot": true, 
        "ami_name": "{{user `aws_ami_name`}}_{{user `aws_ami_version`}}", 
        "ami_description": "Hardened WIndows AMI for Experian AWS Cloud", 
        "vpc_id": "{{ user `vpc_id` }}", 
        "subnet_id": "{{ user `subnet_id` }}",
        "user_data_file": "../../windows/winrm.txt", 
        "ssh_interface": "private_ip", 
        "winrm_username": "Administrator", 
        "temporary_security_group_source_cidrs": ["10.0.0.0/8"],
        "winrm_timeout": "10m", 
        "tags": { 
            "OSVersion": "Windows_2019", 
            "CreatedOn": "{{isotime \"02Jan2006\"}}" 
        }, 
        "source_ami_filter": { 
            "filters": { 
                "virtualization-type": "hvm", 
                "name": "Windows_Server-2019-English-Full-Base*", 
                "root-device-type": "ebs" 
            }, 
            "owners": ["801119661308"], 
            "most_recent": true 
        } 
    }], 
"provisioners": [
    {
            "type": "powershell",
            "inline": [
                "C:\\ProgramData\\Amazon\\EC2-Windows\\Launch\\Scripts\\InitializeInstance.ps1 -SchedulePerBoot",
                "C:\\ProgramData\\Amazon\\EC2-Windows\\Launch\\Scripts\\SysprepInstance.ps1 -NoShutdown"
            ]
        }
]}
```

### Operating system and Environment details

The image is built from a RHEL8 instance using packer 1.8.5 version.

### Log Fragments and crash.log files

Already provided the relevant error above.

**Comment by exolain** [comment]
Adding to my comments above, I re-ran it using the -on-error=abort option to create the image manually.
I do see that sysprep was run correctly, since I see the CloneTag registry key exists:

   CloneTag              : {Tue Dec 27 16:48:06 2022}

Under: HKLM:\System

So not sure what is packer running which is causing the issue just before the image creation.

**Comment by exolain** [comment]
It does seem to be a problem with sysprep itself.
Strange enough, it could be the -NoShutdown flag.

After I made the changes below  Sysprep was successful and the image was created.

Added the "disable_stop_instance": true option to the builder. And then removed the -NoShutdown flag from the Sysprep command, so, it would show just like this.

                "C:\\ProgramData\\Amazon\\EC2-Windows\\Launch\\Scripts\\SysprepInstance.ps1"


I'll close the issue for now. Hopefully it helps anyone going through the same issue
